### ISSUE TITLE
Broken compatibility to Symfony 2.8

### ISSUE THREAD

**Issue by wasinger** [wasinger/htmlpagedom#8]
HtmlPageDom does not work with Symfony 2.8 because of the following changes in the DomCrawler component:

 - Crawler excepts DOMElements only, it's not possible to add text nodes to a Crawler object any more. (See https://github.com/symfony/symfony/pull/16058)

- Cloned Crawler objects don't work due to a PHP Bug (see https://github.com/symfony/symfony/issues/16421)

**Comment by drone076** [comment]
Hi there! Any chances to fix this issue?

**Comment by wasinger** [comment]
Of course I'm planning to make it compatible to Symfony 2.8 and 3, but this will require quite some refactoring and I currently do not have the time to work on this. Maybe some time next near...

**Comment by RubenHarms** [comment]
+1 for this fix!

**Comment by Geczy** [comment]
+1 for symfony 3, can't use Laravel 5.2 without it

**Comment by digilist** [comment]
Are there any updates on this issue? Can you also try to support Symfony 3.0?

Do you have any recent changes? Probably, we can help you out.

**Comment by Geczy** [comment]
Yes it's been three months, have you found any time since then?

**Comment by wasinger** [comment]
Didn't have the time to work on this yet. But pull requests are strongly welcome! Please provide unit tests for all refactorings.

The main challenges are probably:

- Crawler can not contain text nodes any more. That's a problem when working with HTML fragments. So HTML fragment handling probably needs to be reworked.

- Crawler does not extend SplObjectStorage any more, so all methods relying on SplObjectStorage methods need to be refactored.

- As @stof pointed out in https://github.com/symfony/symfony/pull/16058#issuecomment-146474773, we should think about changing the architecture of HtmlPageCrawler to not extend Crawler any more but use decoration pattern

**Comment by wasinger** [comment]
As I noticed only today, since the 2.8.3 release of Symfony HTMLPageDom works with Symfony 2.8! (in 2.8.3, the changes in the 2.8 version of the Symfony DomCrawler component that broke HtmlPageDom have been reverted). I have tagged a new release 1.2.2 that allows Symfony Crawler ^2.8.3 in composer.json. 

Since this issue is titled "Broken compatibility to Symfony 2.8" and the two Symfony issues mentioned in the first post are closed, I will close this issue now. For Symfony 3 compatibility I will open a new one.
